### PR TITLE
Add decision about v2 API

### PR DESCRIPTION
**Pull request by MadVikingGod**
This leaves a record in our document about the findings of V2, and how we won't use a v2.

Closes: #3920 

This does add procedural language about how updates around an API change will happen.  These are my suggestion, and I am open to a different approach.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (dmathieu) [inline CONTRIBUTING.md:492]: This phrase seems unclear to me. Shouldn't it be something like 

```suggestion
Problems happened with libraries that upgraded to v2 when an application did not,
and would not produce any telemetry.
```

**Review by pellared — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Comment by mx-psi** [comment]
I believe this also closes #3277

File: CONTRIBUTING.md
Comment (mx-psi) [inline CONTRIBUTING.md:484]: As has been noted before (e.g. in #3277), this will cause some pain on downstream libraries implementing these interfaces. This may be an acceptable tradeoff (reading through #3920 it certainly seems hard to do this in a painless way for end-users), but I think there should be *some* help from the OTel Go project on how to account for changes to these interfaces in terms of:
1. ensuring these changes are publicly documented in a way that makes it easy for downstream libraries to account for this changes (and if it is just hard for this information to reach the appropriate people, consider if waiting a single version is enough time for downstream users)
2. providing guidance for downstream libraries on how to support multiple versions of the interface (e.g. do we want downstream libraries to have separate implementations for each version of the interfaces as suggested on https://github.com/open-telemetry/opentelemetry-go/issues/3277#issuecomment-1308053982?)

I have not participated in this decision and I am an outsider to OTel Go so I don't know what the proper answer to these questions is, but it feels like before this wording is set in stone they should at least be considered.

Comment (pellared) [reply CONTRIBUTING.md:489]: For metrics API the possibilities are documented here: https://pkg.go.dev/go.opentelemetry.io/otel/metric#hdr-API_Implementations

I think (not validated) that it could be possible to refactor the traces API in a similar way (by introducing `go.opentelemetry.io/otel/trace/embedded`) without introducing breaking changes. However, this would be rather a separate issue.

**Comment by pellared** [comment]
@open-telemetry/go-maintainers Can you please double-check? I am leaning towards merging this PR.

File: CONTRIBUTING.md
Comment (mx-psi) [reply CONTRIBUTING.md:489]: Alright, filed #4160 for this, we can mark this thread as solved. Thanks for following up!

**Review by mx-psi — APPROVED** [review]

**Comment by pellared** [comment]
_Waiting a week before merging._

**Review by dmathieu — APPROVED** [review]

File: CONTRIBUTING.md
Comment (seh) [inline CONTRIBUTING.md:487]:
```suggestion
API change to work seamlessly with the new API.
```

File: CONTRIBUTING.md
Comment (seh) [inline CONTRIBUTING.md:495]:
```suggestion
was no way to introduce a v2 and have it work seamlessly with the v1 of the API.
```

**Pull request merged**
